applications/MedApplication/med_inc.h

```diff
@@ -0,0 +1,21 @@
+// KRATOS  __  __          _    _                _ _           _   _
+//        |  \/  | ___  __| |  / \   _ __  _ __ | (_) ___ __ _| |_(_) ___  _ ___
+//        | |\/| |/ _ \/ _` | / _ \ | '_ \| '_ \| | |/ __/ _` | __| |/ _ \| '_  |
+//        | |  | |  __/ (_| |/ ___ \| |_) | |_) | | | (_| (_| | |_| | (_) | | | |
+//        |_|  |_|\___|\__,_/_/   \_\ .__/| .__/|_|_|\___\__,_|\__|_|\___/|_| |_|
+//                                  |_|   |_|
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher (https://github.com/philbucher)
+//
+
+#pragma once
+
+// Those dummy structs are necessary when the MED-library is compiled with MPI support.
+// Then the header also contains functions which use those MPI-types
```

Comment:
you are right, foward declaring is enough

I used this also in Rust, where the full definitiion was required

Reply:
I'm trying to compile this branch on my system now, and it doesn't work with neither the forward declaration, nor the full definition.

`med.h` includes `hdf5`, which in turn includes `mpi.h` that defines `MPI_Comm` as a `typedef struct` instead of a plain `struct`. Apparently, these are incompatible.

Are you sure you need `MPI_Comm` and `MPI_Info`?

Reply:
discussed in #11580